Formula/smimesign.rb

```diff
@@ -0,0 +1,22 @@
+class Smimesign < Formula
+  desc "S/MIME signing utility for use with Git"
+  homepage "https://github.com/github/smimesign"
+  url "https://github.com/github/smimesign/archive/0.0.3.tar.gz"
+  sha256 "69dfa16737c715679e067d0d929dd964790060b33981695d71bd6b047ae64e97"
+
+  depends_on "go" => :build
+
+  def install
+    ENV["GOPATH"] = buildpath
+    (buildpath/"src/github.com/github/smimesign").install buildpath.children
+
+    cd "src/github.com/github/smimesign" do
+      system "go", "build", "-o", bin/"smimesign", "-ldflags", "-X main.versionString=#{version}"
+      prefix.install_metafiles
+    end
+  end
+
+  test do
+    assert_match version.to_s, shell_output("#{bin}/smimesign --version")
```

Comment:
We need a test that exercises the some of the functionality of the app.

Reply:
The tool doesn't do much unless the user has a signing certificate and key in their keychain. We could just check the exit code from `smimesign --list-keys`, whose output may or may not be empty.

Reply:
Yes, checking the output and exit code would be nice. Possibly try an operation with an invalid key and check that it fails as expected?